src/linux/wpa-controller/ProtocolHostapd.cxx

```diff
@@ -85,3 +86,55 @@ Wpa::WpaPreSharedKeyPropertyKeyAndValue(const WpaPreSharedKey& wpaPreSharedKey)
     auto propertyValue = WpaPreSharedKeyPropertyValue(wpaPreSharedKey);
     return std::make_pair(propertyName, std::move(propertyValue));
 }
+
+WpaKeyManagement
+Wpa::WpaKeyManagementFromPropertyValue(std::string_view wpaKeyManagementProperty) noexcept
+{
+    std::string wpaKeyManagementString(wpaKeyManagementProperty);
+    std::istringstream wpaKeyManagementStream(wpaKeyManagementString);
+    WpaKeyManagement result{ WpaKeyManagement::Unknown };
+
+    for (std::string wpaKeyManagement; wpaKeyManagementStream >> wpaKeyManagement;) {
+        if (wpaKeyManagement == "WPA-EAP") {
+            result = static_cast<WpaKeyManagement>(std::to_underlying(result) | std::to_underlying(WpaKeyManagement::Ieee8021x));
```

Comment:
Converting from an enum's intergral/underlying type should normally be done with [`magic_enum::enum_cast`](https://github.com/Neargye/magic_enum/blob/master/doc/reference.md#enum_cast) instead, since it handles invalid inputs. 

So, you could replace `WpaKeyManagement result{ };` with `std::optional<WpaKeyManagement>`, assign the result of `magic_enum::enum_cast` to it, then check `result.has_value()` at the end.

Though I suspect I'm missing something here. Is there a reason why each one of these is assigning the bitwise-OR of the underlying value of `WpaKeyManagement::Unknown`?

Reply:
`WpaKeyManagement::Unknown` is just `0U`, so that just an empty initialization. In this loop, I'm bitwise-OR'ing each WPA key management value that is found in the property value string, since the string could contain multiple values. I can see if I can still incorporate `magic_enum::enum_cast` though

Reply:
Given my confusion, may also be helpful to add a comment block with the function prototype, and/or a comment at the start roughly describing the approach here.

Reply:
Actually, I just realized my code doesn't work as I thought. I could have a `uint32_t` that is a bitwise-OR of various WPA key management values, but then casting that back to `WpaKeyManagement` wouldn't match any of the values. Perhaps I should store a vector of `WpaKeyManagement` instead